activesupport/lib/active_support/message_encryptor.rb

```diff
@@ -21,7 +21,29 @@ module ActiveSupport
   #   key   = ActiveSupport::KeyGenerator.new('password').generate_key(salt, len) # => "\x89\xE0\x156\xAC..."
   #   crypt = ActiveSupport::MessageEncryptor.new(key)                            # => #<ActiveSupport::MessageEncryptor ...>
   #   encrypted_data = crypt.encrypt_and_sign('my secret data')                   # => "NlFBTTMwOUV5UlA1QlNEN2xkY2d6eThYWWh..."
-  #   crypt.decrypt_and_verify(encrypted_data)                                    # => "my secret data"
+  #   crypt.decrypt_and_verify(encrypted_data) `                                  # => "my secret data"
+  #
+  # MessageEncryptor supports the following metadata that's associated with a message:
+  #
+  # * <tt>:purpose</tt> - The purpose for which the message is intended to be used.
+  # * <tt>:expires_at</tt> - The expiration date of the message.
+  # * <tt>:expires_in</tt> - The relative expiration date. Note that, :expires_at takes
+  #                          precedence over :expires_in.
+  #
+  #   address = "#23 main street"
+  #   encrypted_data = crypt.encrypt_and_sign(address, purpose: "shipping", expires_at: 1.year.from_now) # => "B/yFvIcSzfaVQcjQYU+pGq..."
+  #
+  # Trying to use a message for a different purpose or trying to use an expired message
+  # would fail and return +nil+
+  #
+  #   different_purpose = crypt.decrypt_and_verify(encrypted_data, purpose: "login") # => nil
+  #   address = crypt.decrypt_and_verify(encrypted_data, purpose: :shipping) # => "#23 main street"
```

Comment:
What do you think about removing the variables ? They complicate a bit the example and without them, we can easily see that only the `:purpose` key is different between the two lines.